cmd/system-probe/api/client/client_windows.go

```diff
@@ -8,14 +8,43 @@
 package client
 
 import (
-	"net/http"
+	"context"
+	"fmt"
+	"net"
+	"time"
 
-	processNet "github.com/DataDog/datadog-agent/pkg/process/net"
+	"github.com/Microsoft/go-winio"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	idleConnTimeout = 5 * time.Second
 )
 
-// newSystemProbeClient returns a http client configured to talk to the system-probe
-// This is a simple wrapper around process_net.NewSystemProbeHttpClient because
-// Linux is unable to import pkg/process/net due to size restrictions.
-func newSystemProbeClient(_ string) *http.Client {
-	return processNet.NewSystemProbeClient()
+var (
+	// SystemProbePipeName is the production named pipe for system probe
+	SystemProbePipeName = `\\.\pipe\dd_system_probe`
+)
+
+// DialContextFunc returns a function to be used in http.Transport.DialContext for connecting to system-probe.
+// The result will be OS-specific.
+func DialContextFunc(_ string) func(context.Context, string, string) (net.Conn, error) {
+	return func(_ context.Context, _, _ string) (net.Conn, error) {
+		// Go clients do not immediately close (named pipe) connections when done,
+		// they keep connections idle for a while.  Make sure the idle time
+		// is not too high and the timeout is generous enough for pending connections.
+		var timeout = 30 * time.Second
+
+		namedPipe, err := winio.DialPipe(SystemProbePipeName, &timeout)
+		if err != nil {
+			// This important error may not get reported upstream, making connection failures
+			// very difficult to diagnose. Explicitly log the error here too for diagnostics.
+			var namedPipeErr = fmt.Errorf("error connecting to named pipe %q: %s", SystemProbePipeName, err)
+			log.Error(namedPipeErr.Error())
+			return nil, namedPipeErr
```

Comment:
`log.Error` returns the error, so this can be simplified

```suggestion
			return nil, log.Errorf("error connecting to named pipe %q: %s", SystemProbePipeName, err)
```